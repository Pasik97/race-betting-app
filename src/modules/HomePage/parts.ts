import styled from 'styled-components';

export const WelcomeText = styled.p`
   margin: 0;
   font-size: 18px;
   line-height: 21px;
   text-align: center;
`;
