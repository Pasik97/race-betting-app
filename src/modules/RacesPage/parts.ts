import styled from "styled-components";

export const RacesPageWrapper = styled.div`
    position: relative;
    min-height: 150px;
`;
