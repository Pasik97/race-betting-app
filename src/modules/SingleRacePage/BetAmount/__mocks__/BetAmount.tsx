const mock = () => '(BetAmount /)';

export default mock;
