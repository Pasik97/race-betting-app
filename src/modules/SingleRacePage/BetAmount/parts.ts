import styled from "styled-components";

export const InputWrapper = styled.div`
   display: flex;
   margin: 24px 0;
`;

export const Label = styled.p`
   margin: 0 8px 0 0;
   font-size: 16px;
   line-height: 18px;
   color: #1a1a1a;
`;

export const Input = styled.input`
   color: #1a1a1a;
   font-size: 16px;
   line-height: 21px;
   padding: 0;
   border: none;
   outline: none;
   border-bottom: 1px solid #dddddd;

   &:focus {
      border-bottom: 1px solid #1a1a1a;
   }
`;

