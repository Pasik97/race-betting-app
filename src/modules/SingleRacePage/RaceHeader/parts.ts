import { Link } from 'react-router-dom';
import styled from 'styled-components';

export const RaceTitleWrapper = styled.div`
   display: flex;
   align-items: baseline;
   justify-content: space-between;
`;

export const RaceTitle = styled.h2`
   color: #01263A;
   font-weight: 400;
   font-size: 24px;
   line-height: 27px;
   margin: 16px 0 0 0;
`;

export const RaceTitleStatus = styled.p`
   margin: 0 0 0 8px;
   font-size: 16px; 
   line-height: 18px;
   white-space: nowrap;
`;

export const StyledLink = styled(Link)`
   display: inline-flex;
   align-items: center;
   height: 32px;
   color: #1a1a1a;
   font-size: 13px;
   line-height: 16px;
   text-decoration: none;
   background: #f5f5f5;
   padding: 0px 20px;
   border-radius: 24px;

   &:hover {
      background: #eeeeee;
   }
`;
