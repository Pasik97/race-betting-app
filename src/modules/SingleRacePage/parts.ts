import styled from "styled-components";

export const SingleRacePageWrapper = styled.div`
    position: relative;
    min-height: 150px;
`;
