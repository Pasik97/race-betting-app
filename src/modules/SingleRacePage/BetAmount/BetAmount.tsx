import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import * as P from './parts';
import { setRaceBetAmount } from 'store/races/actions';
import { doubleNumberRegex } from 'utils/validators';
import ApplicationState from 'store/ApplicationState';
import { getRaceBetAmount } from 'store/races/selectors';

interface BetAmountProps {
   raceId: string;
}

const BetAmount: React.FC<BetAmountProps> = (({ raceId }) => {
   const dispatch = useDispatch();

   const raceBetAmount = useSelector((state: ApplicationState) => getRaceBetAmount(state, raceId));

   const [amount, setAmount] = useState<string>(raceBetAmount?.toString() || '0');

   const setBetAmount = (event: React.ChangeEvent<HTMLInputElement>) => {
      const isNumberValid = doubleNumberRegex.test(event.target.value);
      const decimals = event.target.value.split('.')[1];
      const areDecimalsValid = !decimals || decimals.length <= 2;

      if (!event.target.value.length || (isNumberValid && areDecimalsValid)) {
         setAmount(event.target.value);

         const newAmount = Number(event.target.value) || 0;

         dispatch(setRaceBetAmount(raceId, newAmount));
      }
   }

   return (
      <P.InputWrapper>
         <P.Label>Bet amount: </P.Label>
         <P.Input name="betAmount" value={amount} onChange={setBetAmount} />
      </P.InputWrapper>
   )
});

export default BetAmount;
