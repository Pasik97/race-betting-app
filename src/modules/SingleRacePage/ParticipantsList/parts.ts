import styled from 'styled-components';
import { Row } from 'components/BasicTable/parts';
import media from 'utils/media';

export const ParticipantsListWrapper = styled.div``;

export const StyledRow = styled(Row)`
   grid-template-columns: 0.5fr 2fr 0.5fr 0.5fr 0.5fr;

   ${media.fromTablet} {
      grid-template-columns: 0.5fr 4fr 0.5fr 0.5fr 0.5fr;
   }

   ${media.fromDesktop} {
      grid-template-columns: 0.5fr 6fr 0.5fr 0.5fr 0.5fr;
   }
`;

export const Radio = styled.input``;
