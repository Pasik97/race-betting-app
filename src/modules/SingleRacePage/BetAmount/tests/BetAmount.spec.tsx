import { act, create } from "react-test-renderer";
import { Provider } from "react-redux";
import BetAmount from "../BetAmount";
import { Input } from "../parts";
import * as A from 'store/races/actions';
import { createMockRaceStore } from "modules/SingleRacePage/tests/mockData";
import { mockRace, mockRacesState, secondMockRace } from "store/races/tests/mockData";

describe('<BetAmount />', () => {
   it('should render label and input', () => {
      const component = create(
         <Provider store={createMockRaceStore(mockRacesState)}>
            <BetAmount raceId={mockRace.id.toString()} />
         </Provider>,
      );

      expect(component.toJSON()).toMatchSnapshot();
   });

   describe('onAmountChange', () => {
      it('should set amount input value if user enter valid number', () => {
         const component = create(
            <Provider store={createMockRaceStore(mockRacesState)}>
               <BetAmount raceId={secondMockRace.id.toString()} />
            </Provider>,
         );
         let amountInput = component.root.findByType(Input);

         act(() => {
            amountInput.props.onChange({ target: { value: '12.11' } });
         });

         amountInput = component.root.findByType(Input);
   
         expect(amountInput.props.value).toEqual('12.11');
      });

      it('should dispatch setRaceBetAmount if user enter valid number', () => {
         const setRaceBetAmountSpy = jest.spyOn(A, 'setRaceBetAmount');

         const component = create(
            <Provider store={createMockRaceStore(mockRacesState)}>
               <BetAmount raceId={secondMockRace.id.toString()} />
            </Provider>,
         );
         const amountInput = component.root.findByType(Input);
         
         act(() => {
            amountInput.props.onChange({ target: { value: '' } });
         });

         expect(setRaceBetAmountSpy).toHaveBeenCalled();
      });

      it('should not set amount input value if user enter invalid value', () => {
         const component = create(
            <Provider store={createMockRaceStore(mockRacesState)}>
               <BetAmount raceId={secondMockRace.id.toString()} />
            </Provider>,
         );
         let amountInput = component.root.findByType(Input);

         act(() => {
            amountInput.props.onChange({ target: { value: 'awdawd' } });
         });

         amountInput = component.root.findByType(Input);
   
         expect(amountInput.props.value).toEqual('0');
      });

      it('should not set amount input value if decimals in number entered by user are longer than 2', () => {
         const component = create(
            <Provider store={createMockRaceStore(mockRacesState)}>
               <BetAmount raceId={secondMockRace.id.toString()} />
            </Provider>,
         );
         let amountInput = component.root.findByType(Input);

         act(() => {
            amountInput.props.onChange({ target: { value: '12.2222222' } });
         });

         amountInput = component.root.findByType(Input);
   
         expect(amountInput.props.value).toEqual('0');
      });
   });
});
