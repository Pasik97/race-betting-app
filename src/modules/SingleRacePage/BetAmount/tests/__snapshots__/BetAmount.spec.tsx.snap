// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BetAmount /> should render label and input 1`] = `
<div
  className="sc-bdvvtL kFCWXq"
>
  <p
    className="sc-gsDKAQ cjWaGf"
  >
    Bet amount: 
  </p>
  <input
    className="sc-dkPtRN crMWyS"
    name="betAmount"
    onChange={[Function]}
    value="12"
  />
</div>
`;
